bin/reth/src/commands/import_receipts.rs

```diff
@@ -0,0 +1,131 @@
+    datadir: MaybePlatformPath<DataDirPath>,
+
+    /// The chain this node is running.
+    ///
+    /// Possible values are either a built-in chain or the path to a chain specification file.
+    #[arg(
+        long,
+        value_name = "CHAIN_OR_PATH",
+        long_help = chain_help(),
+        default_value = SUPPORTED_CHAINS[0],
+        value_parser = genesis_value_parser
+    )]
+    chain: Arc<ChainSpec>,
+
+    /// Chunk byte length.
+    #[arg(long, value_name = "CHUNK_LEN", verbatim_doc_comment)]
+    chunk_len: Option<u64>,
+
+    #[command(flatten)]
+    db: DatabaseArgs,
+
+    /// The path to a receipts file for import.
+    #[arg(value_name = "IMPORT_PATH", verbatim_doc_comment)]
+    path: PathBuf,
+}
+
+impl ImportReceiptsCommand {
+    /// Execute `import` command
+    pub async fn execute(self) -> eyre::Result<()> {
+        info!(target: "reth::cli", "reth {} starting", SHORT_VERSION);
+
+        debug!(target: "reth::cli",
+            chunk_byte_len=self.chunk_len.unwrap_or(DEFAULT_BYTE_LEN_CHUNK_CHAIN_FILE), "Chunking chain import"
+        );
+
+        // add network name to data dir
+        let data_dir = self.datadir.unwrap_or_chain_default(self.chain.chain);
+
+        let db_path = data_dir.db_path();
+
+        info!(target: "reth::cli", path = ?db_path, "Opening database");
+        let db = Arc::new(init_db(db_path, self.db.database_args())?);
+        info!(target: "reth::cli", "Database opened");
+        let provider_factory =
+            ProviderFactory::new(db.clone(), self.chain.clone(), data_dir.static_files_path())?;
+
+        // open file
+        let mut reader = ChunkedFileReader::new(&self.path, self.chunk_len).await?;
+
+        let tx = provider_factory.provider_rw()?.into_tx();
+        let static_file_provider = provider_factory.static_file_provider();
+
+        let mut total_decoded_receipts = 0;
+
+        while let Some(file_client) = reader.next_chunk::<ReceiptFileClient>().await? {
+            // create a new file client from chunk read from file
+            info!(target: "reth::cli",
+                "Importing receipt file chunk"
+            );
+
+            let ReceiptFileClient { receipts, first_block, total_receipts: total_receipts_chunk } =
+                file_client;
+
+            total_decoded_receipts += total_receipts_chunk;
+
+            let bundled_state =
+                BundleStateWithReceipts::new(Default::default(), receipts, first_block);
+
+            let static_file_producer =
+                static_file_provider.get_writer(first_block, StaticFileSegment::Receipts)?;
+
+            bundled_state.write_to_storage::<<DatabaseEnv as Database>::TXMut>(
+                &tx,
+                Some(static_file_producer),
+                OriginalValuesKnown::Yes,
+            )?;
+        }
+
+        let total_imported_receipts = tx.entries::<tables::Receipts>()?;
+
+        if total_imported_receipts != total_decoded_receipts {
```

Comment:
this should now fail, since `total_imported_receipts` will be 0

Reply:
that surprises me, because this is pretty accurate for blocks and transactions...
```rust
let provider = provider_factory.provider()?;

let total_imported_blocks = provider.tx_ref().entries::<tables::HeaderNumbers>()?;
let total_imported_txns = provider.tx_ref().entries::<tables::TransactionHashNumbers>()?;
```

tested it on https://github.com/paradigmxyz/reth/pull/7781

Reply:
provider is initialised like with path to static files in this pr too

Reply:
When we reference `tables` in the code, it's referring to the DB. Same thing with the `tx_ref` - static files do not have database transactions. The `HeaderNumbers` and `TransactionHashNumbers` tables serve as additional indexes, but nothing like this needs to exist for `Receipts`. So `Receipts` will still return zero here, if we don't insert any receipts

Reply:
thanks for that explanation, will put it in a code comment